### ISSUE TITLE
add more cli tests

### ISSUE THREAD

**Issue by ianstormtaylor** [segmentio/khaos#30]

**Comment by ianstormtaylor** [comment]
@gjohnson @tj @yields @calvinfo could I get some extra input for a sec, I can't figure out how to test the CLI for creation at all. The hard part seems to be faking the user answering questions, trying to write to `stdin` but I can't for the life of me get it to work with a child `spawn`...

Specific code starts here (super hacky cuz I was trying a bunch of stuff, will clean up once I get the thing working): https://github.com/segmentio/khaos/blob/master/test/cli.js#L90-L133

Would love to know if there's just something that I'm doing wrong.

**Comment by gjohnson** [comment]
I generally find if I keep the cli tiny and let commander delegate out to more testable code, that it's not worth the time to test cli flags, maybe that's just me though :-)

**Comment by calvinfo** [comment]
Yeah testing these CLI things is super hard. In herd it was pretty hacky, just ran them as separate processes.

Took a quick look-we'll need to actually be writing `\n` or whatever return character to the `stdin`. Right now emitting the `'keypress`' won't actually do anything unless you setup handlers somewhere else to write it to the process stream. The parent process is keeping a reference to the child's stdin, so emitting won't do anything since the child has no way of knowing `emit` was called–`write` should solve the problem!

**Comment by ianstormtaylor** [comment]
Not testing could always work too haha, I think it'll result in me maintaining less since I'll be more scared of changes though :$ but might have to resort to it

I tried `.write()` as well with line breaks, but I think since the `cli-prompt` uses Nathan's `keypress` under the covers that was screwing me up. I'll try again though. Maybe be able to just write shell scripts that do the running for me with `expect` or something

**Comment by calvinfo** [comment]
Ah yeah. That's kinda what I did with herd: https://github.com/segmentio/herd/blob/master/Makefile#L5-L6

I'd change the answer implementation: https://github.com/segmentio/khaos/blob/master/test/utils.js#L38 to actually write those to the stdin–then it should fix things up.


```
parent: write child.stdin ('\n')
child: process.stdin emits 'keypress' 'enter'
```
